src/key-derivation/hashing-password.md

```diff
@@ -0,0 +1,248 @@
+# Password Hashing
+
+RustCrypto provides a number of [password hash functions], which takes arbitrary data (a password)
+and produces a hash with parameters, which is typically stored in a database. When an unknown
+password is received, it is hashed with the same function and compared with the stored value.
+
+Technically, password hash functions are "key derivation functions" which take a secret key (the
+real password) and turn it into another secret key (the hash stored in the database1).
+
+RustCrypto password hashes implement the [`PasswordHasher`] and [`PasswordVerifier`] traits, which
+provide the hash and check interfaces. These come from the [`password-hash`] create, and are always
+reexported in the hash algorithm crate.
+
+## What should I use?
+
+OWASP puts out a [Password Storage Cheat Sheet], which at time of writing lists the following order
+of preferences:
+
+1. Argon2id with minimum 19 MiB memory, 2 iterations, and 1 degree of parallelism
```

Comment:
Generally numbers one-nine are written out.  Not sure if that holds here.

Reply:
I copied from [here](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html) 🙂 but adjusted it